pinot-core/src/main/java/com/linkedin/pinot/core/query/aggregation/CombineService.java

```diff
@@ -42,14 +43,10 @@
   public static void mergeTwoBlocks(BrokerRequest brokerRequest, IntermediateResultsBlock mergedBlock,
       IntermediateResultsBlock blockToMerge) {
     // Sanity check
+    Preconditions.checkNotNull(mergedBlock);
```

Comment:
are we guaranteed to have non-null mergedBlock ? Even for the first segment returning result ?